### ISSUE TITLE
VB spellchecker crash

### ISSUE THREAD

**Issue by jmarolf** [dotnet/roslyn#6338]
If you have a vb file with the following contents (Assembly level attributes)
```VB
<Assembly: Microsoft.CodeAnalysis.> 
```

VB spellchecker will fail with the following stack

```
	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.SpellCheck.AbstractSpellCheckCodeFixProvider<Microsoft.CodeAnalysis.VisualBasic.Syntax.SimpleNameSyntax>.RegisterCodeFixesAsync(Microsoft.CodeAnalysis.CodeFixes.CodeFixContext context) Line 40	C#
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.CodeFixes.CodeFixService.AppendFixesAsync.AnonymousMethod__2(System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.Diagnostic> dxs) Line 235	C#
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.CodeFixes.CodeFixService.AppendFixesOrSuppressionsAsync.AnonymousMethod__2() Line 280	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions.PerformFunctionAsync<System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.CodeFixes.CodeFix>>(Microsoft.CodeAnalysis.Extensions.IExtensionManager extensionManager, object extension, System.Func<System.Threading.Tasks.Task<System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.CodeFixes.CodeFix>>> function, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.CodeFixes.CodeFix> defaultValue) Line 93	C#
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.CodeFixes.CodeFixService.AppendFixesOrSuppressionsAsync(Microsoft.CodeAnalysis.Document document, Microsoft.CodeAnalysis.Text.TextSpan span, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Diagnostics.DiagnosticData> diagnosticsWithSameSpan, System.Collections.Generic.List<Microsoft.CodeAnalysis.CodeFixes.CodeFixCollection> result, object fixer, System.Func<Microsoft.CodeAnalysis.Diagnostic, bool> hasFix, System.Func<System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.Diagnostic>, System.Threading.Tasks.Task<System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.CodeFixes.CodeFix>>> getFixes, System.Threading.CancellationToken cancellationToken) Line 280	C#
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.CodeFixes.CodeFixService.AppendFixesAsync(Microsoft.CodeAnalysis.Document document, Microsoft.CodeAnalysis.Text.TextSpan span, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Diagnostics.DiagnosticData> diagnostics, System.Collections.Generic.List<Microsoft.CodeAnalysis.CodeFixes.CodeFixCollection> result, System.Threading.CancellationToken cancellationToken) Line 240	C#
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.CodeFixes.CodeFixService.GetFixesAsync(Microsoft.CodeAnalysis.Document document, Microsoft.CodeAnalysis.Text.TextSpan range, bool includeSuppressionFixes, System.Threading.CancellationToken cancellationToken) Line 140	C#
 	[Resuming Async Method]	
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.MoveNextRunner.InvokeMoveNext(object stateMachine) Line 1090	C#
 	mscorlib.dll!System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx) Line 954	C#
 	mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx) Line 902	C#
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.MoveNextRunner.Run() Line 1070	C#
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.OutputAsyncCausalityEvents.AnonymousMethod__0() Line 977	C#
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.ContinuationWrapper.Invoke() Line 1123	C#
 	mscorlib.dll!System.Runtime.CompilerServices.TaskAwaiter.OutputWaitEtwEvents.AnonymousMethod__0() Line 288	C#
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.ContinuationWrapper.Invoke() Line 1123	C#
 	mscorlib.dll!System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(System.Action action, bool allowInlining, ref System.Threading.Tasks.Task currentTask) Line 810	C#
 	mscorlib.dll!System.Threading.Tasks.Task.FinishContinuations() Line 3617	C#
 	mscorlib.dll!System.Threading.Tasks.Task.FinishStageThree() Line 2363	C#
 	mscorlib.dll!System.Threading.Tasks.Task<System.__Canon>.TrySetResult(System.__Canon result) Line 490	C#
 	mscorlib.dll!System.Threading.Tasks.UnwrapPromise<System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Diagnostics.DiagnosticData>>.TrySetFromTask(System.Threading.Tasks.Task task, bool lookForOce) Line 7272	C#
 	mscorlib.dll!System.Threading.Tasks.UnwrapPromise<System.__Canon>.ProcessInnerTask(System.Threading.Tasks.Task task) Line 7293	C#
 	mscorlib.dll!System.Threading.Tasks.UnwrapPromise<System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Diagnostics.DiagnosticData>>.ProcessCompletedOuterTask(System.Threading.Tasks.Task task) Line 7222	C#
 	mscorlib.dll!System.Threading.Tasks.UnwrapPromise<System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Diagnostics.DiagnosticData>>.Invoke(System.Threading.Tasks.Task completingTask) Line 7146	C#
 	mscorlib.dll!System.Threading.Tasks.Task.FinishContinuations() Line 3626	C#
 	mscorlib.dll!System.Threading.Tasks.Task.FinishStageThree() Line 2363	C#
 	mscorlib.dll!System.Threading.Tasks.Task.FinishStageTwo() Line 2336	C#
 	mscorlib.dll!System.Threading.Tasks.Task.Finish(bool bUserDelegateExecuted) Line 2253	C#
 	mscorlib.dll!System.Threading.Tasks.Task.ExecuteWithThreadLocal(ref System.Threading.Tasks.Task currentTaskSlot) Line 2830	C#
 	mscorlib.dll!System.Threading.Tasks.Task.ExecuteEntry(bool bPreventDoubleExecution) Line 2767	C#
 	mscorlib.dll!System.Threading.Tasks.Task.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem() Line 2704	C#
 	mscorlib.dll!System.Threading.ThreadPoolWorkQueue.Dispatch() Line 820	C#
 	mscorlib.dll!System.Threading._ThreadPoolWaitCallback.PerformWaitCallback() Line 1161	C#

```

**Comment by davkean** [comment]
We should figure out if this is a regression, if so, this makes the stabilization bar.

**Comment by jmarolf** [comment]
bug was introduced with this commit: https://github.com/dotnet/roslyn/commit/6190e895e9a8823184aab45f8455d36eef3d4c5f on September 12th 2015

**Comment by jmarolf** [comment]
previously we would check if the node was missing: https://github.com/dotnet/roslyn/commit/6190e895e9a8823184aab45f8455d36eef3d4c5f#diff-3ccebe3c9c23f44a544a9ae888e48605L64

**Comment by Pilchie** [comment]
Fixed by #6343
